front/app/components/InitiativeMap/index.tsx

```diff
@@ -0,0 +1,251 @@
+  const startInitiativeButtonNode = document.createElement('div');
+  const initiativeInfoNode = document.createElement('div');
+  initiativeInfoNode.id = 'initiative-info-node';
+
+  // Loop through initatives with locations and create array of graphics
+  const initiativesWithLocation = list.filter(
+    (initiative) => initiative?.attributes?.location_point_geojson
+  );
+  const graphics = initiativesWithLocation?.map((initiative) => {
+    return new Graphic({
+      geometry: new Point({
+        longitude:
+          initiative?.attributes?.location_point_geojson?.coordinates[0],
+        latitude:
+          initiative?.attributes?.location_point_geojson?.coordinates[1],
+      }),
+      attributes: {
+        initiativeId: initiative?.id,
+      },
+    });
+  });
+
+  // Create an Esri Layer from the graphics so we can add a cluster display
+  const initiativesLayer = graphics
+    ? new FeatureLayer({
+        source: graphics, // Array of initiative graphics
+        title: 'Initiative pins',
+        objectIdField: 'ID',
+        fields: [
+          {
+            name: 'ID',
+            type: 'oid',
+          },
+          {
+            name: 'initiativeId', // From the graphics attributes
+            type: 'string',
+          },
+        ],
+        // Set the symbol used to render the graphics
+        renderer: new Renderer({
+          symbol: getMapPinSymbol(theme.colors.tenantPrimary),
+        }),
+        // Add cluster display to this layer
+        featureReduction: getClusterConfiguration(theme.colors.tenantPrimary),
+      })
+    : undefined;
+
+  const onMapClick = (event: any, mapView: MapView) => {
+    // On map click, we either open an existing initiative or show the "submit a proposal" popup
+    mapView.hitTest(event).then((result) => {
+      const elements = result.results; // These are elements under our map click
+      if (elements.length > 0) {
+        // User clicked an initiative pin OR cluster
+        elements.forEach((element) => {
+          if (element.type === 'graphic') {
+            const graphicId = element.graphic.attributes.ID;
+            if (graphicId === undefined) {
+              // User clicked a cluster. Zoom in on the cluster.
+              mapView.goTo({
+                center: [element.mapPoint.longitude, element.mapPoint.latitude],
+                zoom: mapView.zoom + 1,
+              });
+            } else {
+              // User clicked an initiative pin
+              const initiativeId = graphics?.at(graphicId - 1)?.attributes
+                .initiativeId;
+              if (initiativeId) {
+                mapView.goTo({
+                  center: [
+                    element.mapPoint.longitude,
+                    element.mapPoint.latitude,
+                  ],
+                });
+                // Emit an event so we show the selected initiative's information
+                eventEmitter.emit('initiativeSelectedEvent', initiativeId);
+              }
```

Comment:
Using an eventEmitter to pass the selected initiative ID to the initiative preview component was the only way I found which didn't cause the Esri Map to re-render (which causes a brief white screen + resets the view). I know we don't usually want eventEmitters, but this was the only way I found to get this working correctly :thinking: